dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -94,13 +94,15 @@ tasks.withType<Test>().configureEach {
     }
 
     // environment() isn't Provider API compatible yet https://github.com/gradle/gradle/issues/11534
-    dokkaBuild.integrationTestExhaustive.orNull?.let { exhaustive ->
-        environment("isExhaustive", exhaustive)
-    }
-    dokkaBuild.androidSdkDir.orNull?.let { androidSdkDir ->
-        environment("ANDROID_HOME", androidSdkDir.invariantSeparatorsPath)
+    fun environmentProvider(name: String, provider: Provider<out Any>) {
+        inputs.property(name, provider)
```

Comment:
@adam-enko FYI, without adding properties which are passed to `environment` to `inputs` tasks will always stay `up-to-date` when those properties are changed

Reply:
Huh, that's weird! I guess that's a consequence of the configuration cache.

Looks like it might be fixed in https://github.com/gradle/gradle/issues/11534 (finally!)

In the meantime we could convert them to set using CommandLineArgumentProvider instead - but in another PR.